rotkehlchen/tests/utils/exchanges.py

```diff
@@ -823,3 +836,46 @@ def try_get_first_exchange(
         return None
 
     return exchanges_list[0]
+
+
+def mock_exchange_data_in_db(exchange_locations, rotki) -> None:
+    db = rotki.data.db
+    for exchange_location in exchange_locations:
+        db.add_trades([Trade(
+            timestamp=Timestamp(1),
+            location=exchange_location,
+            base_asset=A_BTC,
+            quote_asset=A_ETH,
+            trade_type=TradeType.BUY,
+            amount=AssetAmount(FVal(1)),
+            rate=Price(FVal(1)),
+            fee=Fee(FVal('0.1')),
+            fee_currency=A_ETH,
+            link='foo',
+            notes='boo',
+        )])
+        db.update_used_query_range(name=f'{str(exchange_location)}_trades_{str(exchange_location)}', start_ts=0, end_ts=9999)  # noqa: E501
+        db.update_used_query_range(name=f'{str(exchange_location)}_margins_{str(exchange_location)}', start_ts=0, end_ts=9999)  # noqa: E501
+        db.update_used_query_range(name=f'{str(exchange_location)}_asset_movements_{str(exchange_location)}', start_ts=0, end_ts=9999)  # noqa: E501
+
+
+def check_saved_events_for_exchange(
+        exchange_location: Location,
+        db: DBHandler,
+        should_exist: bool,
+        query_range_key: str = '{exchange}_{type}_{exchange}',
```

Comment:
name this queryrange_formatstr, as it's meant to not be passed as a formatstr. Spent quite some time figuring this out